### ISSUE TITLE
Inaccurate search result

### ISSUE THREAD

**Issue by EnixCoda** [EnixCoda/Gitako#224]
When searching `tooltip` in `https://github.com/ant-design/ant-design/blob/3.x-stable/components/tooltip/index.tsx`, only part of the content of the `/components/tooltip` folder showed up. Files like `/components/tooltip/index.tsx` is missing.

<img width="1025" alt="image" src="https://user-images.githubusercontent.com/7480839/161365076-4007a85c-002a-413c-b609-bd642f34953d.png">

**Comment by EnixCoda** [comment]
Looks like it is a feature 😅

How it works:
When searching `tooltip`, both the `/components/tooltip` dir and its inner `/components/tooltip/__tests__/__snapshots__/tooltip.test.js` match. And when there are inner items match, not all items of the dir will be displayed.

Such strategy was designed for a cleaner search result. But now it confuses its designer myself.

**Comment by EnixCoda** [comment]
Well. Perhaps this could be a better solution to the problem.

<img width="338" alt="image" src="https://user-images.githubusercontent.com/7480839/163719750-0d28ddd1-cd54-462a-996d-21f1c468239c.png">

Instead omitting all unmatched items, all dirs that have no items match will be collapsed. The search result will still be clean in most cases and should not introduce such confusion.

**Comment by EnixCoda** [comment]
😅 Found another serious problem of search, some files may be missing from search results.

<img width="606" alt="image" src="https://user-images.githubusercontent.com/7480839/164960082-5e887d7d-5d12-4a84-9e62-2924ea689c9d.png">

The items in green background are the missing ones. Some of them, which have no highlight characters(`b`) are included for search logic change. But others are expected to be always included.

Caused by the `g` flag used in regular expressions.

```
> r = /./g;
[
    r.test('d'),
    r.test('d'),
    r.test('d'),
    r.test('d'),
]
< [true, false, true, false]
```


```
> r = /./;
[
    r.test('d'),
    r.test('d'),
    r.test('d'),
    r.test('d'),
]
< [true, true, true, true]
```

**Comment by EnixCoda** [comment]
Fixed in d9272a7b24144d9b3c6bb37faa2914adf8a3c365
